internal/cluster/server/server_test.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	protoc "github.com/pbinitiative/zenbpm/internal/cluster/command/proto"
+	"github.com/pbinitiative/zenbpm/internal/cluster/network"
+	"github.com/pbinitiative/zenbpm/internal/cluster/proto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestServer(t *testing.T) {
+	ctx := t.Context()
+
+	mux, err := network.NewMux("")
+	if err != nil {
+		t.Fatalf("failed to create new mux: %s", err)
+	}
+	cLn := network.NewZenBpmClusterListener(mux)
+	tStore := &testStore{}
+	srv := New(cLn, tStore)
+	err = srv.Open()
+	if err != nil {
+		t.Fatalf("failed to start server: %s", err)
+	}
+
+	dialer := network.NewZenBpmClusterDialer()
+	fmt.Println(cLn.Addr().String())
+	grpcClient, err := grpc.NewClient(cLn.Addr().String(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+			return dialer.Dial(s, 1*time.Second)
+		}),
+	)
+	if err != nil {
+		t.Fatalf("failed to create grpc client: %s", err)
+	}
+	zsc := proto.NewZenServiceClient(grpcClient)
+
+	_, err = zsc.Notify(ctx, &proto.NotifyRequest{
+		Id:      "123",
+		Address: "local-1.cluster",
+	})
+	if err != nil {
+		t.Fatalf("failed to notify server: %s", err)
+	}
+	if tStore.notify == nil || tStore.notify.Id != "123" {
+		t.Fatalf("unexpected notify result")
+	}
+
+	_, err = zsc.Join(ctx, &proto.JoinRequest{
+		Id:      "123",
+		Address: "local-1.cluster",
+		Voter:   true,
+	})
+	if err != nil {
+		t.Fatalf("failed to notify server: %s", err)
+	}
+	if tStore.join == nil || tStore.join.Id != "123" || tStore.join.Address != "local-1.cluster" {
```

Comment:
This could be rewritten like the following to ease understanding of the assertion
```go
then.AssertThat(t, tStore.join, is.Nil())
then.AssertThat(t, tStore.join.Id, is.EqualTo("123"))
then.AssertThat(t, tStore.join.Address, is.EqualTo("local-1.cluster"))
```